### DISCUSSION TITLE
How to hide {password} used in scenario steps in Report

### DISCUSSION THREAD

**Discussion by vickymore2003** [WasiqB/multiple-cucumber-html-reporter#555] [Q&A] [answered]
I have a below scenario for login

Given user login in with "<username>" and "<password>"
example
username | password
Jsmith      | Tiger@123

when test is run and report is generated i get below string in report
user login in with "Jsmith" and "Tiger@123"

how can i hide the password and replace it with ***** in report

**Comment by AbdulkadirC** [comment]
Depends on which language you use. Here I leave a sample from my (Typescript) framework:
`console.log(`${key} is found for ${userType}: ${key === "wachtwoord" ? "***" : userData[key]}`);`

**Comment by WasiqB** [comment, marked as answer]
Hi @vickymore2003 , @AbdulkadirC ,

I would recommend not to pass sensitive data to the `attach` method in the first place.
If it is required to show these sensitive information, then it is suggested that you handle the masking of the data at your end.

In the meantime, I would suggest to try using the new [4.0.0-beta.0](https://github.com/WasiqB/multiple-cucumber-html-reporter/releases/tag/v4.0.0-beta.0) version which was released yesterday.

You can check out the [attachment related documentation](https://multiple-cucumber-html-reporter.vercel.app/docs/attachments) for more information.
